Sources/MediaViewController.swift

```diff
@@ -164,25 +185,46 @@ class VLCMediaViewController<T>: UICollectionViewController, UICollectionViewDel
     }
 
     override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        if let playlistCell = collectionView.dequeueReusableCell(withReuseIdentifier: VLCPlaylistCollectionViewCell.cellIdentifier(), for: indexPath) as? VLCPlaylistCollectionViewCell {
-            if let mediaObject = subcategory.files[indexPath.row] as? NSManagedObject {
-                playlistCell.mediaObject = mediaObject
+        if isEditing {
```

Comment:
could we set a different datasource when we enter editmode to avoid the if else?

Reply:
Indeed it sounds nicer! Should it be a custom dataSource? 🤔

Reply:
yes a custom object. you can take a look at the mediaviewcontroller where we switch between searchdatasource and the usual one